droid-hal-device.inc

```diff
@@ -829,20 +785,67 @@ fi
 %endif
 %endif
 
+%pre users
+#Compatibility with old droid-hal version - convert previous format
+#of droid-specific user/group information to current one.
+cd %{_libdir}/droid
+if [ $1 -gt 1 -a \
+     -e droid-user-remove.sh.installed ]; then
+  echo '#!/bin/sh' > droid-user-remove.sh.installed
+fi
+
+%post users
+cd %{_libdir}/droid
+
+# NOTE: We do not delete groups/users in any case since that would cause issues
+# with users being removed from groups where they should really be in.
+
+# Now for both install/update add the users and mark which were added already
+echo creating droid users and groups
+# On fresh installs, the file is not there at all and would break grep below
+DROID_USERS=$(cat droid.ids)
+if [ "$DROID_USERS" != "" ]; then
+    echo "$DROID_USERS" |while read name id; do
```

Comment:
This could be simplified to just
`[ -e droid.ids ] && cat droid.ids | while read name id; do`